securedrop_client/gui/state.py

```diff
@@ -0,0 +1,44 @@
+import textwrap
+from typing import Dict, List, NewType, Optional
+from uuid import UUID
+
+from PyQt5.QtCore import QObject
+
+ConversationId = NewType("ConversationId", UUID)
+FileId = NewType("FileId", UUID)
+
+
+class State(QObject):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self._selected_conversation: Optional[ConversationId] = None
+        self._conversations: Dict[ConversationId, List[FileId]] = {}
+
+    def selected_conversation(self) -> Optional[ConversationId]:
+        return self._selected_conversation
+
+    def conversation_files(self, id: ConversationId) -> List[FileId]:
+        return self._conversations.get(id, [])
+
+    def __repr__(self) -> str:
+        return textwrap.dedent(
+            f"""
+                selected_conversation: {self._selected_conversation}
+                conversation_files: {self._conversations}
+            """
+        )
+
+
+class Writer:
+    def __init__(self, state: State) -> None:
+        self._state = state
+
+    def set_selected_conversation(self, id: ConversationId) -> None:
+        if id not in self._state._conversations:
+            self.upsert_conversation(id, [])
```

Comment:
we should only set the selected conversation if it exists, so you can remove this if clause

Reply:
Done.